src/net.md

```diff
@@ -876,6 +877,86 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-check-broken-links]: #ex-check-broken-links
+<a name="ex-check-broken-links"/>
+## Check a webpage for broken links
+
+[![reqwest-badge]][reqwest] [![select-badge]][select] [![url-badge]][url] [![cat-net-badge]][cat-net]
+
+We call "get_base_url" to retrieve the base URL. If the document has a "base" tag, we get the "href" [`attr`] from the first occurrence of the "base" tag. This is then used as the base URL. Otherwise, we can use [`Position::BeforePath`] with the original URL to get the base of that URL.
+
+We iterate through all the links in the document and get the absolute URL for each (using [`url::ParseOptions`] and [`Url::parse`]). We then filter these so that we can report which links are broken.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+extern crate select;
+extern crate url;
+
+use url::{Url, Position};
+use reqwest::StatusCode;
+use select::document::Document;
+use select::predicate::Name;
+
+use std::collections::HashSet;
+#
+# error_chain! {
+#   foreign_links {
+#       ReqError(reqwest::Error);
+#       IoError(std::io::Error);
+#       UrlParseError(url::ParseError);
+#   }
+# }
+fn get_base_url(url: &Url, doc: &Document) -> Result<Url> {
+    let base_tag_href = doc.find(Name("base")).filter_map(|n| n.attr("href")).nth(0);
+
+    let base_url = base_tag_href.map_or_else(
+        || Url::parse(&url[..Position::BeforePath]),
+        Url::parse,
+    )?;
+
+    Ok(base_url)
+}
+
+fn check_link(url: &Url) -> Result<bool> {
+    let res = reqwest::get(url.as_ref())?;
+
+    Ok(res.status() != StatusCode::NotFound)
+}
+
+fn run() -> Result<()> {
+    let url = Url::parse("https://www.rust-lang.org/en-US/")?;
+
+    let res = reqwest::get(url.as_ref())?;
+    let document = Document::from_read(res)?;
+
+    let base_url = get_base_url(&url, &document)?;
+
+    let base_parser = Url::options().base_url(Some(&base_url));
+
+    let links: HashSet<Url> = document
+        .find(Name("a"))
+        .filter_map(|n| n.attr("href"))
+        .filter_map(|link| base_parser.parse(link).ok())
+        .collect();
+
+    let broken_links = links.iter().filter(|link| {
```

Comment:
no need for intermediate `broken_links` variable. we can iterate over the filtered iterator in a for loop